### ISSUE TITLE
Test action fails when `path` isn't set on Kustomization resource

### ISSUE THREAD

**Issue by rtrox** [allenporter/flux-local#237]
In [the spec](https://fluxcd.io/flux/components/kustomize/api/v1/) for the `Kustomization` kind in `kustomize.toolkit.fluxcd.io/v1`, the `path` field is optional:

>(Optional)
> Path to the directory containing the kustomization.yaml file, or the set of plain YAMLs a kustomization.yaml should be generated for. Defaults to ‘None’, which translates to the root path of the SourceRef.

However, when the test action is run against a repo using kustomizations with `path` unset, `flux-local` throws an input exception:
https://github.com/allenporter/flux-local/blob/60be4ba89b8028f3253e9c320ac642e2d8a56083/flux_local/manifest.py#L295-L296

Example manifests which trigger this error:
```yaml
---
apiVersion: source.toolkit.fluxcd.io/v1beta1
kind: GitRepository
metadata:
  name: kube-prometheus-stack-source
  namespace: flux-system
spec:
  interval: 30m
  url: https://github.com/prometheus-community/helm-charts.git
  ref:
    tag: kube-prometheus-stack-43.0.0
  ignore: |
    # exclude all
    /*
    # include deploy crds dir
    !/charts/kube-prometheus-stack/crds
---
apiVersion: kustomize.toolkit.fluxcd.io/v1
kind: Kustomization
metadata:
  name: kube-prometheus-stack-crds
  namespace: flux-system
spec:
  interval: 15m
  prune: false
  sourceRef:
    kind: GitRepository
    name: kube-prometheus-stack-source
  healthChecks:
    - apiVersion: apiextensions.k8s.io/v1
      kind: CustomResourceDefinition
      name: alertmanagerconfigs.monitoring.coreos.com
    - apiVersion: apiextensions.k8s.io/v1
      kind: CustomResourceDefinition
      name: alertmanagers.monitoring.coreos.com
    - apiVersion: apiextensions.k8s.io/v1
      kind: CustomResourceDefinition
      name: podmonitors.monitoring.coreos.com
    - apiVersion: apiextensions.k8s.io/v1
      kind: CustomResourceDefinition
      name: probes.monitoring.coreos.com
    - apiVersion: apiextensions.k8s.io/v1
      kind: CustomResourceDefinition
      name: prometheuses.monitoring.coreos.com
    - apiVersion: apiextensions.k8s.io/v1
      kind: CustomResourceDefinition
      name: prometheusrules.monitoring.coreos.com
    - apiVersion: apiextensions.k8s.io/v1
      kind: CustomResourceDefinition
      name: servicemonitors.monitoring.coreos.com
    - apiVersion: apiextensions.k8s.io/v1
      kind: CustomResourceDefinition
      name: thanosrulers.monitoring.coreos.com
```
Source: https://github.com/rtrox/home-cluster/blob/main/cluster-cd/crds/kube-prometheus-stack/crds.yaml

Full Logs from run with traceback:
```
Run flux-local \
  flux-local \
    --log-level INFO \
    test \
    --enable-helm \
    --no-enable-kyverno \
    --api-versions "policy/v1/PodDisruptionBudget" \
    --kustomize-build-flags="" \
    --sources "" \
    --path cluster-cd/clusters/chongus
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.10.12/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.10.12/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.10.12/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.10.12/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.10.12/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.10.12/x64/lib
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/runner/work/_actions/allenporter/flux-local/main/flux_local/git_repo.py", line 368, in kustomization_traversal
INTERNALERROR>     docs = await get_flux_kustomizations(root, path)
INTERNALERROR>   File "/home/runner/work/_actions/allenporter/flux-local/main/flux_local/git_repo.py", line 329, in get_flux_kustomizations
INTERNALERROR>     return [
INTERNALERROR>   File "/home/runner/work/_actions/allenporter/flux-local/main/flux_local/git_repo.py", line 330, in <listcomp>
INTERNALERROR>     Kustomization.parse_doc(doc)
INTERNALERROR>   File "/home/runner/work/_actions/allenporter/flux-local/main/flux_local/manifest.py", line 296, in parse_doc
INTERNALERROR>     raise InputException(f"Invalid {cls} missing spec.path: {doc}")
INTERNALERROR> flux_local.exceptions.InputException: Invalid <class 'flux_local.manifest.Kustomization'> missing spec.path: {'apiVersion': 'kustomize.toolkit.fluxcd.io/v1', 'kind': 'Kustomization', 'metadata': {'name': 'kube-prometheus-stack-crds', 'namespace': 'flux-system', 'annotations': {'config.kubernetes.io/index': '1', 'config.kubernetes.io/path': 'kube-prometheus-stack/crds.yaml', 'internal.config.kubernetes.io/index': '1', 'internal.config.kubernetes.io/path': 'kube-prometheus-stack/crds.yaml'}}, 'spec': {'interval': '15m', 'prune': False, 'sourceRef': {'kind': 'GitRepository', 'name': 'kube-prometheus-stack-source'}, 'healthChecks': [{'apiVersion': 'apiextensions.k8s.io/v1', 'kind': 'CustomResourceDefinition', 'name': 'alertmanagerconfigs.monitoring.coreos.com'}, {'apiVersion': 'apiextensions.k8s.io/v1', 'kind': 'CustomResourceDefinition', 'name': 'alertmanagers.monitoring.coreos.com'}, {'apiVersion': 'apiextensions.k8s.io/v1', 'kind': 'CustomResourceDefinition', 'name': 'podmonitors.monitoring.coreos.com'}, {'apiVersion': 'apiextensions.k8s.io/v1', 'kind': 'CustomResourceDefinition', 'name': 'probes.monitoring.coreos.com'}, {'apiVersion': 'apiextensions.k8s.io/v1', 'kind': 'CustomResourceDefinition', 'name': 'prometheuses.monitoring.coreos.com'}, {'apiVersion': 'apiextensions.k8s.io/v1', 'kind': 'CustomResourceDefinition', 'name': 'prometheusrules.monitoring.coreos.com'}, {'apiVersion': 'apiextensions.k8s.io/v1', 'kind': 'CustomResourceDefinition', 'name': 'servicemonitors.monitoring.coreos.com'}, {'apiVersion': 'apiextensions.k8s.io/v1', 'kind': 'CustomResourceDefinition', 'name': 'thanosrulers.monitoring.coreos.com'}]}}
INTERNALERROR> 
INTERNALERROR> During handling of the above exception, another exception occurred:
INTERNALERROR> 
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/_pytest/main.py", line 267, in wrap_session
INTERNALERROR>     config.hook.pytest_sessionstart(session=session)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/pluggy/_hooks.py", line 265, in __call__
INTERNALERROR>     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/pluggy/_manager.py", line 80, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/pluggy/_callers.py", line 60, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/pluggy/_result.py", line 60, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/pluggy/_callers.py", line 39, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/runner/work/_actions/allenporter/flux-local/main/flux_local/tool/test.py", line 280, in pytest_sessionstart
INTERNALERROR>     asyncio.run(self.async_pytest_sessionstart(session))
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/nest_asyncio.py", line 35, in run
INTERNALERROR>     return loop.run_until_complete(task)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/nest_asyncio.py", line 90, in run_until_complete
INTERNALERROR>     return f.result()
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/asyncio/futures.py", line 201, in result
INTERNALERROR>     raise self._exception.with_traceback(self._exception_tb)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/asyncio/tasks.py", line 232, in __step
INTERNALERROR>     result = coro.send(None)
INTERNALERROR>   File "/home/runner/work/_actions/allenporter/flux-local/main/flux_local/tool/test.py", line 285, in async_pytest_sessionstart
INTERNALERROR>     manifest = await git_repo.build_manifest(
INTERNALERROR>   File "/home/runner/work/_actions/allenporter/flux-local/main/flux_local/git_repo.py", line 656, in build_manifest
INTERNALERROR>     clusters = await get_clusters(
INTERNALERROR>   File "/home/runner/work/_actions/allenporter/flux-local/main/flux_local/git_repo.py", line 534, in get_clusters
INTERNALERROR>     kustomizations = await kustomization_traversal(path_selector)
INTERNALERROR>   File "/home/runner/work/_actions/allenporter/flux-local/main/flux_local/git_repo.py", line 371, in kustomization_traversal
INTERNALERROR>     raise FluxException(
INTERNALERROR> flux_local.exceptions.FluxException: Error building Fluxtomization in '/home/runner/work/home-cluster/home-cluster' path 'cluster-cd/crds': Invalid <class 'flux_local.manifest.Kustomization'> missing spec.path: {'apiVersion': 'kustomize.toolkit.fluxcd.io/v1', 'kind': 'Kustomization', 'metadata': {'name': 'kube-prometheus-stack-crds', 'namespace': 'flux-system', 'annotations': {'config.kubernetes.io/index': '1', 'config.kubernetes.io/path': 'kube-prometheus-stack/crds.yaml', 'internal.config.kubernetes.io/index': '1', 'internal.config.kubernetes.io/path': 'kube-prometheus-stack/crds.yaml'}}, 'spec': {'interval': '15m', 'prune': False, 'sourceRef': {'kind': 'GitRepository', 'name': 'kube-prometheus-stack-source'}, 'healthChecks': [{'apiVersion': 'apiextensions.k8s.io/v1', 'kind': 'CustomResourceDefinition', 'name': 'alertmanagerconfigs.monitoring.coreos.com'}, {'apiVersion': 'apiextensions.k8s.io/v1', 'kind': 'CustomResourceDefinition', 'name': 'alertmanagers.monitoring.coreos.com'}, {'apiVersion': 'apiextensions.k8s.io/v1', 'kind': 'CustomResourceDefinition', 'name': 'podmonitors.monitoring.coreos.com'}, {'apiVersion': 'apiextensions.k8s.io/v1', 'kind': 'CustomResourceDefinition', 'name': 'probes.monitoring.coreos.com'}, {'apiVersion': 'apiextensions.k8s.io/v1', 'kind': 'CustomResourceDefinition', 'name': 'prometheuses.monitoring.coreos.com'}, {'apiVersion': 'apiextensions.k8s.io/v1', 'kind': 'CustomResourceDefinition', 'name': 'prometheusrules.monitoring.coreos.com'}, {'apiVersion': 'apiextensions.k8s.io/v1', 'kind': 'CustomResourceDefinition', 'name': 'servicemonitors.monitoring.coreos.com'}, {'apiVersion': 'apiextensions.k8s.io/v1', 'kind': 'CustomResourceDefinition', 'name': 'thanosrulers.monitoring.coreos.com'}]}}Is a Kustomization pointing to a path that does not exist?
```
Github Actions run: https://github.com/rtrox/home-cluster/actions/runs/5299408363/jobs/9592345745

**Comment by allenporter** [comment]
Thank you for the excellent bug report. I'll work on a fix shortly.

**Comment by allenporter** [comment]
I have a pending PR which seems to address this and i tried it out on your cluster:
```
$ flux-local test --path cluster-cd/clusters/chongus/
======================================= test session starts ========================================
collected 9 items                                                                                  

cluster-cd/clusters/chongus .........                                                        [100%]

======================================== 9 passed in 6.62s =========================================
```

**Comment by allenporter** [comment]
Fixed in 1.3.3

**Comment by rtrox** [comment]
Thanks @allenporter, this is great! I'll try adding it into CI this weekend!
